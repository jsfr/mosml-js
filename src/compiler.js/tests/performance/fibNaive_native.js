"use strict"
var ctimeBegin$1 = function($_var1) {
    return console.time($_var1);
};
var ctimeEnd$2 = function($_var5) {
    return console.timeEnd($_var5);
};
var repeat$3 = function($_var9) {
    return function($_var10) {
        return function($_var11) {
            var $_var13 = function($_var14) {
                return (function() {
                    switch ($_var14) {
                        case 0:
                            return $_var10($_var11)
                        default:
                            return ($_var10($_var11), $_var13($_var14 - 1))
                    }
                }());
            };
            return $_var13($_var9 - 1);
        };
    };
};
var timerep$4 = function($_var35) {
    return function($_var36) {
        return function($_var37) {
            return repeat$3($_var35)(function($_var41) {
                return (ctimeBegin$1($_var37), ($_var36($_var41), ctimeEnd$2($_var37)));
            });
        };
    };
};

var naiveFib = function(n) {
  switch(n)
  {
    case 0.0:
      return 1.0;

    case 1.0:
      return 1.0;

    default:
      return naiveFib(n-1.0) + naiveFib(n-2.0)
    }
}

var $_var66 = timerep$4(100)(naiveFib)("naiveFib_native")(35.0);