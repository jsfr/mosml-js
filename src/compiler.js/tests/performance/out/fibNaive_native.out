==============================JS BEGIN==============================
naiveFib_native: 147ms
naiveFib_native: 146ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 154ms
naiveFib_native: 155ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 179ms
naiveFib_native: 157ms
naiveFib_native: 148ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 153ms
naiveFib_native: 148ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 149ms
naiveFib_native: 169ms
naiveFib_native: 151ms
naiveFib_native: 148ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 202ms
naiveFib_native: 155ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 176ms
naiveFib_native: 155ms
naiveFib_native: 151ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 148ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 169ms
naiveFib_native: 171ms
naiveFib_native: 152ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 148ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 153ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 182ms
naiveFib_native: 151ms
naiveFib_native: 153ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 146ms
naiveFib_native: 147ms
naiveFib_native: 147ms
naiveFib_native: 146ms
naiveFib_native: 153ms
naiveFib_native: 146ms
naiveFib_native: 158ms
naiveFib_native: 155ms
naiveFib_native: 144ms
==============================JS END================================
